addons/test/misra/misra-test.c

```diff
@@ -234,10 +234,17 @@ void misra_9_5() {
   int x[] = {[0]=23}; // 9.5
 }
 
-void misra_10_1(uint8_t u) {
+void misra_10_1(uint8_t u, char c1, char c2) {
   int32_t i;
+  char c;
+  enum { E1 = 1 };
   i = 3 << 1; // 10.1
   i = (u & u) << 4; // no-warning
+  c = c1 & c2; // FIXME: this is not compliant to "10.1"
+  c = c1 << 1; // 10.1
+  i = c1 - c2; // FIXME: false positive for 10.6 this is compliant
```

Comment:
is there something wrong with Travis? If there is a false positive anywhere in `misra-test.c` then Travis should fail as far as I know. Is it silenced by the FIXME comment?

It also seems a bit misplaced. It should be added in `misra_10_6`.

Reply:
Travis is ok 🙂 I created the PR so it does not fail. It just adds FIXME comments.
The false positive is silenced by the `10.6` that is part of the comment.
The false negative is silenced by the `"` around `10.1`. Thus the rule number is not detected.

This is an example for rule `10.1` and listed in the documentation and in the example suite under rule `10.1`. So IMO it is not misplaced.
It could be copied to `misra_10_6`, maybe as part of the fix to prevent a regression i suggest.